code/game/objects/items/toys.dm

```diff
@@ -691,19 +689,49 @@
 	if(cards.len == 0)
 		to_chat(user, "<span class='warning'>There are no more cards to draw!</span>")
 		return
-	var/obj/item/toy/cards/singlecard/H = new/obj/item/toy/cards/singlecard(user.loc)
-	if(holo)
-		holo.spawned += H // track them leaving the holodeck
-	choice = cards[1]
-	H.cardname = choice
-	H.parentdeck = src
-	var/O = src
-	H.apply_card_vars(H,O)
-	src.cards.Cut(1,2)
-	H.pickup(user)
-	user.put_in_hands(H)
-	user.visible_message("[user] draws a card from the deck.", "<span class='notice'>You draw a card from the deck.</span>")
-	update_icon()
+	if (drawnumber==1)
+		var/obj/item/toy/cards/singlecard/C = new/obj/item/toy/cards/singlecard(user.loc)
+		choice = cards[1]
+		user.visible_message("[user] draws a card from the deck.", "<span class='notice'>You draw a card from the deck.</span>")
+		C.cardname = choice
+		if(holo)
+			holo.spawned += C // track them leaving the holodeck
+		C.parentdeck = src
+		var/O = src
+		C.apply_card_vars(C,O)
+		C.deckstyle=deckstyle
+		src.cards.Cut(1,2)
+		C.pickup(user)
+		user.put_in_hands(C)
+		update_icon()
+		C.interact(user)
+	else //if more than one card is drawn
+		var/obj/item/toy/cards/cardhand/H = new/obj/item/toy/cards/cardhand(user.loc)
+		user.visible_message("[user] draws [drawnumber] cards from the deck.", "<span class='notice'>You draw [drawnumber] cards from the deck.</span>")
+		var/i
+		for (i=1,i<=drawnumber,i++)
+			H.currenthand+=cards[i]
+		if(holo)
+			holo.spawned += H // track them leaving the holodeck
+		H.parentdeck = src
+		H.deckstyle=deckstyle
+		src.cards.Cut(1,drawnumber+1)
+		H.pickup(user)
+		user.put_in_hands(H)
+		update_icon()
+		H.interact(user)
+		H.update_icon()
+
+/obj/item/toy/cards/deck/AltClick(mob/living/L)
+	if(!(L.mobility_flags & MOBILITY_PICKUP))
+		return
+	if(cards.len == 0)
+		to_chat(L, "<span class='warning'>There are no more cards to draw!</span>")
+		return
+	var/drawsize = stripped_input(L, "How many cards to draw? (1-[min(cards.len,10)])", "Cards")
```

Comment:
You dont need a stripped input here, you can use a normal input() as num | null.
Its going to show a proper number input too

Reply:
ok!